src/Config.js

```diff
@@ -0,0 +1,61 @@
+'use strict';
+require('regenerator-runtime/runtime');
+const _ = require('lodash');
+const DataSourceManager = require('./DataSourceManager');
+
+class Config {
+  constructor(dataSources) {
+    this.dataSources = dataSources;
+  }
+
+  _buildDataSourceManagers() {
```

Comment:
If this is synchronous, why not wrap it into the constructor?

Reply:
It's called from get(key) as well, below.

Reply:
Right, that's my point. Put it in the constructor and no one else has to worry about it.

Reply:
Ah, got it.  One goal here was to vend async versions of functions so that, when clients adopt async/await, they can get the benefits of lazy initialization.  There may be, for example, lambda clients who want the bugsnag infrastructure but not any of this config stuff, and if they don't call "await get" they won't hit persistent storage and load those plugins.

I'm not strongly opinionated here though.  If you feel the code isn't clean and would rather have the initialization up-front, I'm ok with that.  I don't think this is like, a huge win, just a small optimization.

Reply:
Right, I got that intention, and I think it's an elegant solution. My point is more narrow, which is that `_buildDataSourceManagers` has to happen before the client can use _either_ the new or legacy get(), and it only synchronously creates the DataSourceManagers rather than actually initializing them, so it could be deduped and moved to the constructor.